src/cluster/k8sClientHandler.go

```diff
@@ -581,26 +581,24 @@ func GetNodesFromK8sClient() (*v1.NodeList, error) {
 }
 
 func GetKubearmorRelayURL() string {
-	var namespace string
 	client := ConnectK8sClient()
 	if client == nil {
 		return ""
 	}
 
 	// get kubearmor-relay pod from k8s api client
-	pods, err := client.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+	svc, err := client.CoreV1().Services("").List(context.Background(), metav1.ListOptions{
 		LabelSelector: "kubearmor-app=kubearmor-relay",
```

Comment:
are we using this label for kubearmor service?
```
k -n kube-system get svc --show-labels | grep -i kubearmor
kubearmor                                       ClusterIP   10.0.45.5     <none>        32767/TCP       47h   <none>
kubearmor-annotation-manager-metrics-service    ClusterIP   10.0.178.96   <none>        443/TCP         47h   kubearmor-app=kubearmor-annotation-manager
kubearmor-host-policy-manager-metrics-service   ClusterIP   10.0.67.88    <none>        8443/TCP        47h   kubearmor-app=kubearmor-host-policy-manager
kubearmor-policy-manager-metrics-service        ClusterIP   10.0.155.61   <none>        8443/TCP        47h   kubearmor-app=kubearmor-policy-manager
```

Reply:
![image](https://github.com/accuknox/discovery-engine/assets/89014588/85507b03-1e80-46ea-8372-192fbeb9b372)
I think we are using **kubearmor-app=kubearmor-relay** as the label @vishnusomank

Reply:
this label is for the service pod selector, right? meaning this svc will redirect traffic to the pod with label `kubearmor-app=kubearmor-relay`. I don't think we have attached a label to the kubearmor svc.
```
k -n kube-system get svc kubearmor -o yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2023-06-21T06:41:49Z"
  name: kubearmor
  namespace: kube-system
  resourceVersion: "8642527"
  uid: 77ab50bf-fec9-4d16-af37-f10fe53bbc6b
spec:
  clusterIP: 10.0.45.5
  clusterIPs:
  - 10.0.45.5
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - port: 32767
    protocol: TCP
    targetPort: 32767
  selector:
    kubearmor-app: kubearmor-relay
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

Reply:
ig `LabelSelector` is used to look for labels under metadata and not `Spec.Selector`. 
`kubearmor-app=kubearmor-relay` is the value of `selector` in `kubearmor` service.
```
apiVersion: v1
kind: Service
metadata:
  name: kubearmor
  namespace: kube-system
spec:
  ports:
  - port: 32767
    protocol: TCP
    targetPort: 32767
  selector:
    kubearmor-app: kubearmor-relay
```
@rajaSahil PTAL at this issue.